core/templates/pages/teach-page/teach-page.component.spec.ts

```diff
@@ -15,112 +15,193 @@
 /**
  * @fileoverview Unit tests for the teach page.
  */
-import { WindowRef } from 'services/contextual/window-ref.service';
 
-require('pages/teach-page/teach-page.component.ts');
+import { ComponentFixture, fakeAsync, TestBed, tick } from
+  '@angular/core/testing';
+import { NO_ERRORS_SCHEMA, Pipe, EventEmitter } from '@angular/core';
 
-describe('Teach Page', function() {
-  var $scope = null, ctrl = null;
-  var $timeout = null;
-  var SiteAnalyticsService = null;
-  var windowRef = new WindowRef();
-
-  beforeEach(angular.mock.module('oppia'));
-  beforeEach(angular.mock.module('oppia', function($provide) {
-    $provide.value('WindowRef', windowRef);
-  }));
-  beforeEach(angular.mock.inject(function($injector, $componentController) {
-    $timeout = $injector.get('$timeout');
-    SiteAnalyticsService = $injector.get('SiteAnalyticsService');
+import { TeachPageComponent } from './teach-page.component';
+import { I18nLanguageCodeService } from 'services/i18n-language-code.service';
+import { SiteAnalyticsService } from 'services/site-analytics.service';
+import { TranslateService } from 'services/translate.service';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
 
-    var $rootScope = $injector.get('$rootScope');
-    $scope = $rootScope.$new();
+// Mocking window object here because changing location.href causes the
+// full page to reload. Page reloads raise an error in karma.
+
+@Pipe({name: 'translate'})
+class MockTranslatePipe {
+  transform(value: string, params: Object | undefined):string {
+    return value;
+  }
+}
+
+class MockTranslateService {
+  languageCode = 'es';
+  use(newLanguageCode: string): string {
+    this.languageCode = newLanguageCode;
+    return this.languageCode;
+  }
+}
+
+class MockI18nLanguageCodeService {
+  codeChangeEventEmiiter = new EventEmitter<string>();
+  getCurrentI18nLanguageCode() {
+    return 'en';
+  }
+
+  get onI18nLanguageCodeChange() {
+    return this.codeChangeEventEmiiter;
+  }
+}
```

Comment:
Add a line at the beginning of each class.

Reply:
Done